### DISCUSSION TITLE
Add known/standard metadata keys to documentation

### DISCUSSION THREAD

**Discussion by jphastings** [cooklang/spec#101] [General]
It'd be great if the [metadata section](https://cooklang.org/docs/spec/#metadata) of the docs was expanded to list all the known/standard keys and their intended purpose & format.

Right now it's not clear if we should make our own if they're unlisted (eg. `servings` isn't mentioned there, but it appears to be considered standard from other pages in the docs).

What I'm assuming the list would be so far:

| Key | Purpose | Example value |
| --- | --- | --- |
| `source` | Where the recipe came from. Usually a URL, can also be text (eg. a book title) with anything hidden from display (ie. for machine use) in angle brackets afterwards. | `https://example.org/recipe`, `The Palomar Cookbook <urn:isbn:9781784720995>` |
| `time required` | The preparation + cook time of the recipe. Various formats can be parsed, if in doubt use `HhMm` format to avoid plurals and locales. | `45 minutes`, `1 hour 30 minutes`,`1h30m` |
| `servings` | Indicates how many people the recipe is for Used for scaling quantities. Leading number is used for scaling, anything else is ignored but shown as units. | `2`,`15 cups worth` |
| `course` | ??? I'm not sure! This looks like it'd hold a category name, rather than a course, especially when multilingual stuff comes into play? | `dinner` |
| `locale` | The locale of the recipe. Used for spelling/grammar during edits, and for pluralisation of amounts. Uses ISO 639 language code, then optionally an underscore and the ISO 3166 alpha2 "country code" for dialect variants | `es_VE`, `en_GB`, `fr`  |

I'd love to have clarity on these as I'd like to know where I can put information in software I'm writing such that other cooklang tools can interpret it correctly.

For example, if I'm turning a recipe from a physical book into a cooklang recipe I'd like to be able to store the ISBN and the page number(s). I'd be tempted to put that like this, but I'd want to know that (a) the name of the source would be used and (b) the URN would be hidden (possibly [linked to](https://openlibrary.org/works/OL20052517W/The_Palomar_cookbook?edition=key:/books/OL27232537M) for clients that support ISBN URNs) (edit: it looks like this is assumed by the [cooklang extensions](https://github.com/cooklang/cooklang-rs/blob/main/extensions.md), can I consider them standard?)

```
source: The Palomar Cookbook <urn:isbn:9781784720995#pages=42-43>
```

I'd like to propose the following keys too:

| Key | Purpose | Example value |
| --- | --- | --- |
| `cook time` | The time it take to just cook the recipe. Formatted as `time required` field | `4 hours`, `4h` |
| `prep time` | The time it take to just prep the recipe. Formatted as `time required` field | `15 minutes`,`1h30m` |

I'm new here, but I'll be happy to make a PR if this makes sense.

**Comment by jphastings** [comment]
I hadn't spotted that this has been partially done with the [cooklang rs extensions](https://github.com/cooklang/cooklang-rs/blob/main/extensions.md) — it'd be great if this was referenced too!

**Comment by dubadub** [comment]
Relevant proposal for spec change https://github.com/cooklang/spec/pull/105.

**Comment by dubadub** [comment]
This is now part of the main specification, closing the discussion.
